spec/index.bs

```diff
@@ -3582,21 +3597,40 @@ The {{GPUMapMode}} flags determine how a {{GPUBuffer}} is mapped when calling
         Only valid with buffers created with the {{GPUBufferUsage/MAP_READ}} usage.
 
         Once the buffer is mapped, calls to {{GPUBuffer/getMappedRange()}} will return an
-        {{ArrayBuffer}} containing the buffer's current values. Changes to the returned
-        {{ArrayBuffer}} will be discarded after {{GPUBuffer/unmap()}} is called.
+        {{ArrayBuffer}} containing the buffer's current values.
+
+        Changes to the returned {{ArrayBuffer}} will be discarded after {{GPUBuffer/unmap()}} is
+        called.
 
     : <dfn>WRITE</dfn>
     ::
         Only valid with buffers created with the {{GPUBufferUsage/MAP_WRITE}} usage.
 
+        When the buffer is created without [=GPU-writable buffer usages=]:
+        Once the buffer is mapped, calls to {{GPUBuffer/getMappedRange()}} will return an
+        {{ArrayBuffer}} containing the buffer’s current values.
+
+        When the buffer is created with [=GPU-writable buffer usages=]:
         Once the buffer is mapped, calls to {{GPUBuffer/getMappedRange()}} will return an
-        {{ArrayBuffer}} containing the buffer's current values. Changes to the returned
-        {{ArrayBuffer}} will be stored in the {{GPUBuffer}} after {{GPUBuffer/unmap()}} is called.
+        {{ArrayBuffer}} containing the default initialized data (zeros) or data written by the
+        webpage during a previous mapping.
+
+        Changes to the returned {{ArrayBuffer}} will be stored in the buffer after
+        {{GPUBuffer/unmap()}} is called.
+
+        Note: Write-only mapping will never return values produced by the GPU.
```

Comment:
Why is this restriction necessary? Won't it result in an extra copy of the data for write only mappings on UMA systems?

I think it would be preferable to state that reading the contents of a write only mappings are undefined in this case.

Reply:
> Won't it result in an extra copy of the data for write only mappings on UMA systems?
It is mainly because currently in Chromium a copy is always needed because we cannot access the mapped pointer from GPU resources directly in the JS side. We can discuss more about this issue in the WG meeting.

Reply:
We have the same restriction in WebKit too, perhaps this is a non-issue.

Reply:
> I think it would be preferable to state that reading the contents of a write only mappings are undefined (or zero-ed?) in this case.

We couldn't make it fully undefined, at most we could add a third option for what gets returned. Hard to say if the performance would be any better better if we allowed it. It would require cache flushes or something to make sure you don't get unsafe undefined values.

And of course for portability reasons we really don't want anyone to rely on data getting read back on write-only mappings. But that's probably not a _huge_ concern.